shared/js/l10n.js

```diff
@@ -1607,27 +1684,32 @@
       }
     }
 
-    if (!this.ctx.availableLocales.length) {
-      // if there was no availableLanguages meta,
-      // register the default locale only
-      this.ctx.registerLocales(this.ctx.defaultLocale);
+    if (navigator.mozApps && navigator.mozApps.getAdditionalLanguages) {
+      navigator.mozApps.getAdditionalLanguages().then(
+        registerLocales.bind(this, meta));
+    } else {
+      registerLocales.call(this, meta);
     }
-    return initLocale.call(this);
+  }
+
+  function registerLocales(meta, extraLangs) {
+    this.ctx.registerLocales(meta.defaultLanguage,
+                             meta.availableLanguages,
+                             extraLangs);
+    initLocale.call(this);
   }
 
   function splitAvailableLanguagesString(str) {
-    return str.split(',').map(function(lang) {
+    var langs = {};
+
+    str.split(',').forEach(function(lang) {
       lang = lang.trim().split(':');
-      // if there are no timestamps, lang[0] will be the ab-CD
-      return lang[0];
+      langs[lang[0]] = lang[1];
     });
+    return langs;
   }
 
-  function onMetaInjected(node) {
-    if (this.ctx.availableLocales.length) {
-      return;
-    }
-
+  function onMetaInjected(node, meta) {
```

Comment:
That's really a consistency discussion. I want to keep all onElementInjected in bindings have similar signature, and from that perspective meta is an additional extra param. I prefer to keep it that way.